### PR TITLE
bug/feat/BUILT-73_Client_Profile_Update_Sync

### PR DESCRIPTION
**Pull request by Chloedunwoody**
JIRA: [BUILT-73](https://champlainsaintlambert.atlassian.net/browse/BUILT-73?atlOrigin=eyJpIjoiYzdhNDIyYzZkY2FhNDZkZWI4MDIyMmJmMDgxOTE2N2YiLCJwIjoiaiJ9)

## Context
This bug fix aims to fix the issue with syncing the profile data once it has been updated.

## Changes
- Refresh data method 
- update test case

### CODE REVIEW

File: BuiltDifferentMobileApp/BuiltDifferentMobileApp/ViewModels/Profile/MyProfilePageClientViewModel.cs [hunk 1]
```diff
@@ -68,19 +68,24 @@ public bool IsEnabled
...
             CurrentWeight = user.currentWeight;
-
-            SubmitCommand = new AsyncCommand(Submit);
-            UploadImageCommand = new AsyncCommand(Upload);
-            EditProfileCommand = new AsyncCommand(Edit);
-
-            isEnabled = false;
         }
 
         private async Task Edit()
```
Comment (bthorng) [inline BuiltDifferentMobileApp/BuiltDifferentMobileApp/ViewModels/Profile/MyProfilePageClientViewModel.cs:91]: Everything looks good, good job!
While we're here, I'm just wondering for the profile there is an edit button and you can click it to switch between edit and non-edit mode, correct?

Comment (Chloedunwoody) [reply BuiltDifferentMobileApp/BuiltDifferentMobileApp/ViewModels/Profile/MyProfilePageClientViewModel.cs:91]: yes correct, ill double check to make sure its in working order

File: BuiltDifferentMobileApp/BuiltDifferentMobileApp/ViewModels/Profile/MyProfilePageClientViewModel.cs [hunk 2]
```diff
@@ -68,19 +68,24 @@ public bool IsEnabled
...
+            UploadImageCommand = new AsyncCommand(Upload);
+            EditProfileCommand = new AsyncCommand(Edit);
+
+            isEnabled = false;
+        }
+
+        private async Task GetUserInfo()
+        {
             Models.Client user = (Models.Client)accountService.CurrentUser;
             Id = user.id;
```
Comment (bthorng) [inline BuiltDifferentMobileApp/BuiltDifferentMobileApp/ViewModels/Profile/MyProfilePageClientViewModel.cs:83]: From what I understand, GetUserInfo() retrieves the CurrentUser from the account service and sets the inputs on the screen to it's properties. But I don't see anything updating the current user stored in the AccountService.
Consider fetching the current user's profile after updating it and and setting it to .CurrentUser in AccountService and then setting the form fields.
Pseudo code:

`var userInfo = networkService.GetAsync<>(CURRENT_USER_PROFILE_URI);`
// you can check how I set the current user in the network service register function
`accountService.CurrentUser = userInfo;`

Let me know if I'm wrong.

File: BuiltDifferentMobileApp/BuiltDifferentMobileApp/ViewModels/Profile/MyProfilePageClientViewModel.cs (hunk 1 above)
Comment (bthorng) [reply BuiltDifferentMobileApp/BuiltDifferentMobileApp/ViewModels/Profile/MyProfilePageClientViewModel.cs:91]: Yes and there's a confirm/cancel button correct?
If so, be careful about user's inputting changes and cancelling their changes, you will need to set the form's fields to their previous values.

File: BuiltDifferentMobileApp/BuiltDifferentMobileApp/ViewModels/Profile/MyProfilePageClientViewModel.cs (hunk 2 above)
Comment (Chloedunwoody) [reply BuiltDifferentMobileApp/BuiltDifferentMobileApp/ViewModels/Profile/MyProfilePageClientViewModel.cs:89]: changes have been made to reflect these suggestions. Thank you so much !

File: BuiltDifferentMobileApp/BuiltDifferentMobileApp/ViewModels/Profile/MyProfilePageClientViewModel.cs (hunk 1 above)
Comment (Chloedunwoody) [reply BuiltDifferentMobileApp/BuiltDifferentMobileApp/ViewModels/Profile/MyProfilePageClientViewModel.cs:97]: Changes have been made for this suggestion, will reretrieve the profile if not saved. (reseting the profile data

File: BuiltDifferentMobileApp/BuiltDifferentMobileApp/ViewModels/Profile/MyProfilePageClientViewModel.cs (hunk 2 above)
Comment (Chloedunwoody) [reply BuiltDifferentMobileApp/BuiltDifferentMobileApp/ViewModels/Profile/MyProfilePageClientViewModel.cs]: You were correct, the changes have been rectified.

**Comment by bthorng** [comment]
Everything looks good now from what I can tell, feel free to merge if you are done and everything works.

If you want to be more efficient you can store the previous values for the fields so that if a user cancels their edit instead of fetching from the API, you just revert it back to what you stored in memory (this is just a nitpick tho)
Also don't forget to remove your test variable before merging
Line 149: `var test = JsonConvert.SerializeObject(profile);`

**Pull request merged**